### PR TITLE
feat: persistance of login and singup forms

### PR DESCRIPTION
**Pull request by ahtesham-quraish**
### Description

- Persistence of login, signup forgot password forms
- Error alert should not be visible on forms switches

#### JIRA

[VAN-1957](https://2u-internal.atlassian.net/browse/VAN-1957)
[VAN-2007](https://2u-internal.atlassian.net/browse/VAN-2007)

#### How Has This Been Tested?

It has been tested locally

### CODE REVIEW

**Review by mubbsharanwar — COMMENTED** [review]
Have we tested the flow when the pipeline is active and the user switch between login and register forms that the form data should be presist?

File: src/forms/login-popup/data/reducers.js [hunk 1]
```diff
@@ -59,6 +67,9 @@ export const loginSlice = createSlice({
     setLoginSSOIntent: (state) => {
       state.isLoginSSOIntent = true;
     },
+    backupLoginFormBegin: (state, { payload }) => {
```
[the remark this answers is not in the document]
Comment (ahtesham-quraish) [reply src/forms/login-popup/data/reducers.js:70]: Done

**Comment by ahtesham-quraish** [comment]
> Have we tested the flow when the pipeline is active and the user switch between login and register forms that the form data should be presist?

yes it is working i have added the change

File: src/forms/registration-popup/data/reducers.js [hunk 2]
```diff
@@ -65,6 +73,9 @@ export const registerSlice = createSlice({
     setRegistrationFields: (state, { payload }) => {
       state.registrationFields = payload;
     },
+    backupRegistrationFormBegin: (state, { payload }) => {
```
[the remark this answers is not in the document]
Comment (ahtesham-quraish) [reply src/forms/registration-popup/data/reducers.js:77]: done

File: src/forms/registration-popup/tests/RegistrationPopup.test.jsx [hunk 3]
```diff
@@ -183,6 +193,20 @@ describe('RegistrationForm Test', () => {
     expect(store.dispatch).not.toHaveBeenCalledWith(registerUser({}));
   });
 
+  it('should backup the registration form state when switch to login form', () => {
+    store = mockStore({
+      ...initialState,
+      register: {
+        ...initialState.register,
+      },
+    });
```
[the remark this answers is not in the document]
Comment (ahtesham-quraish) [reply src/forms/registration-popup/tests/RegistrationPopup.test.jsx:202]: It does not work that is why i have added these lines

File: src/forms/registration-popup/data/reducers.js [hunk 4]
```diff
@@ -16,6 +16,15 @@ export const registerInitialState = {
   validationState: DEFAULT_STATE,
   registrationError: {},
   registrationResult: {},
+  registrationFormData: {
+    isFormDirty: false,
```
Comment (mubbsharanwar) [inline src/forms/registration-popup/data/reducers.js:20]:
```suggestion
    isFormFilled: false,
```

**Review by mubbsharanwar — APPROVED** [review]

**Pull request merged**